.github/workflows/hamilton-ui-build-and-push.yml

```diff
@@ -0,0 +1,65 @@
+name: Building and pushing UI frontend and backend images
+
+on:
+  schedule:
+    - cron: '0 0 * * *'  
+  workflow_dispatch:  
+
+jobs:
+  check_and_build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - name: Install jq
+        run: sudo apt-get update && sudo apt-get install -y jq
+
+      - name: Check latest version of sf-hamilton-ui
+        id: check_version
+        run: |
+          response=$(curl -s https://pypi.org/pypi/sf-hamilton-ui/json)
+          version=$(echo "$response" | jq -r '.info.version')
+          echo "latest_version=$version" >> $GITHUB_ENV
+
+      - name: Restore previous version from cache
+        id: restore_cache
+        uses: actions/cache@v4
+        with:
+          path: ./version_cache
+          key: hamilton-ui-version-cache
+
+      - name: Compare versions
+        id: compare_versions
+        run: |
+          PREVIOUS_VERSION=$(cat version_cache/previous_version.txt 2>/dev/null || echo "none")
+          echo "Previous version: $PREVIOUS_VERSION"
+          if [[ "$latest_version" == "$PREVIOUS_VERSION" ]]; then
+            echo "No new version found. Exiting."
+            exit 0
+          else
+            echo "New version detected: $latest_version"
+          fi
```

Comment:
you could probably instead check what's pushed to docker hub instead? i.e. what is the highest version value? That might be more robust?

Reply:
> you could probably instead check what's pushed to docker hub instead? i.e. what is the highest version value? That might be more robust? Thoughts?
Well, In that case, I might not be able to check if the previous version(for the previous build)  is the same as the current version, and then would not be able to early exit if they're the same. So, that is why I've used the cache action.
The latest version is being obtained from the sf-hamilton-ui package that's there on PyPI .